maintenance_request_purchase/models/maintenance_request.py

```diff
@@ -23,6 +23,28 @@ class MaintenanceRequest(models.Model):
         groups="purchase.group_purchase_user",
     )
 
+    total_purchase_amount = fields.Monetary(
+        compute="_compute_total_purchase_amount",
+        store=True,
+        groups="purchase.group_purchase_user",
+    )
+
+    currency_id = fields.Many2one(
+        "res.currency", string="Currency", compute="_compute_currency", store=True
+    )
```

Comment:
Could you simply try this instead of a computed method?
```suggestion
    currency_id = fields.Many2one(
        related="company_id.currency_id", store=True
    )
```

Reply:
done 👍🏽